### PR TITLE
fix Hammerhead initialization in iframe (close #184)

### LINKED ISSUE #184: Hammerhead must be initialized in iframe synchoronously
Content for reproduce:
### index.html
```javascript
<html>
	<body>
	    <script>
			window.isIframeInitializedSynchronously = false;
		</script>
		<iframe id="iframe" src="/iframe.html"></iframe>
		<script>
		    var iframe = document.getElementById('iframe');
			iframe.addEventListener('load', function() {
                //////////Must be 'true'
				console.log('isIframeInitializedSynchronously - %s', window.isIframeInitializedSynchronously);
			});
		</script>
	</body>
</html>
```

### iframe.html
```javascript
<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <title></title>
    <meta charset="utf-8">
</head>
<body>
    <script>
        window.top.isIframeInitializedSynchronously = true;
    </script>
    <h1>This is iframe</h1>
</body>
</html>
```

### PR DESCRIPTION
**Pull request by miherlosev**
/cc @inikulin @churkin

### CODE REVIEW

File: src/client/transport.js [hunk 1]
```diff
@@ -144,77 +141,81 @@ class Transport extends EventEmitter {
         });
     }
 
-    asyncServiceMsg (msg) {
-        return new Promise(resolve => {
-            msg.sessionId = settings.get().sessionId;
+    performRequest (msg, callback) {
```
Comment (inikulin) [inline src/client/transport.js]: Add comment here, with `TODO` what we need to turn it back to the Promise, once we get rid of `syncServiceMsg`

File: test/client/fixtures/sandbox/iframe-test.js [hunk 2]
```diff
@@ -43,6 +43,22 @@ test('event should not raise if a cross-domain iframe is appended', function ()
     $iframe.remove();
 });
 
+asyncTest('iframe initialization must be synchronously', function () {
```
Comment (inikulin) [inline test/client/fixtures/sandbox/iframe-test.js]: Check text with @VasilyStrelyaev. Not all iframes should be initialized sync, it should be in title. It should be in regression section with appropriate `GH-{num}` postfix

**Comment by inikulin** [comment]
\r-

File: src/client/transport.js [hunk 3]
```diff
@@ -144,77 +141,81 @@ class Transport extends EventEmitter {
...
+                request.addEventListener('readystatechange', function () {
+                    if (this.readyState !== 4)
+                        return;
+
+                    msgCallback();
+                });
+            }
+            else if (isIE9) {
+                //aborted ajax request in IE9 not raise error, abort or timeout events
+                //also getting status code raise error c00c023f
```
Comment (VasilyStrelyaev) [inline src/client/transport.js]: Aborted ajax requests do not raise the error, abort or timeout events in IE9. Getting a status code raises the c00c023f error.

File: test/client/fixtures/sandbox/iframe-test.js (hunk 2 above)
Comment (VasilyStrelyaev) [inline test/client/fixtures/sandbox/iframe-test.js]: discussed in person

**Comment by miherlosev** [comment]
FPR

**Comment by VasilyStrelyaev** [comment]
lgtm

File: test/client/fixtures/sandbox/iframe-test.js [hunk 4]
```diff
@@ -163,3 +163,19 @@ asyncTest('the AMD module loader damages proxing an iframe without src (GH-127)'
     });
     document.body.appendChild(iframe);
 });
+
+asyncTest('iframe initialization must be synchronous (except for iframes with an empty src) (GH-184)', function () {
```
Comment (inikulin) [inline test/client/fixtures/sandbox/iframe-test.js]: Wait, AFAIK it should be sync ONLY for the iframes with empty src

**Comment by inikulin** [comment]
\r-

**Comment by miherlosev** [comment]
FPR

**Comment by inikulin** [comment]
lgtm

**Comment by VasilyStrelyaev** [comment]
lgtm

**Comment by churkin** [comment]
lgtm

**Pull request merged**